### PR TITLE
SJ WORKFLOW ENHANCEMENTS. As an open-source user, I don't want to get caught out by dead ends while creating my pipeline so that I have a smooth productive journey.

### PR DESCRIPTION
**Pull request by richardmatthewsdev**
**Acceptance Criteria**
- User can easily start transformation after completing extraction.
- User can easily start an optional enrichment
- The relevant opportunities to improve user experience during the pipeline workflow are incorporated.
- User inputs are reviewed and if possible auto populated with a useful default, or removed.
- Users focus is directed to relevant controls

### CODE REVIEW

**Review by motizuki — APPROVED** [review]
Looks good overall, but the PR was quite large for me to do an in-depth review, do you have any areas in the PR you would like some specific feedback?

**Comment by richardmatthewsdev** [comment]
> Looks good overall, but the PR was quite large for me to do an in-depth review, do you have any areas in the PR you would like some specific feedback?

Thanks @motizuki! It is sort of a grab bag of changes from the user testing so that's fair enough. More of a sanity check really :).

File: app/views/pipelines/_enrichment_definition.html.erb
Comment (lun-boost) [inline app/views/pipelines/_enrichment_definition.html.erb:1]: I'm struggling a little bit to read this file, would you be able to comment some chunks of code?

File: app/views/transformation_definitions/show.html.erb [hunk 1]
```diff
@@ -26,7 +22,31 @@
...
 
-<div id="js-transformation-app" data-props="<%= @props %>"></div>
+<% if @transformation_definition.extraction_job.completed? %>
+  <div id="js-transformation-app" data-props="<%= @props %>"></div>
+<% else %>
+  <div class="container">
+    <div class="row">
+      <div class="col text-center">
+          <p class='text-center'>Waiting for extraction to run. Please wait a minute and refresh the page.</p>
+          <%= link_to '', class: 'btn btn-outline-primary' do %>
```
Comment (lun-boost) [inline app/views/transformation_definitions/show.html.erb:32]: Does this refresh the page? That's cool :O

**Review by lun-boost — APPROVED** [review]
Nice one Richard, looks mighty fine!! 🌟 It's a little bit challenging to review without having much context/knowledge of Supplejack Harvester but so many new stuff for me that I'm interested to learn more about. Don't need to reply to my comments, let's catch up in the afternoon instead!!

**Pull request merged**